### PR TITLE
KOGITO-7766 - Add Data Index Addon example using PostgreSQL persistence

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

File: serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# Kogito Serverless Workflow - Data Index Example
+
+## Description
+
+This example contains a simple workflow service that demonstrates how to use Data Index Addon as part of the Kogito runtime. 
+A callback is a state that invokes an action and wait for an event (event that will be eventually fired by the external service notified by the action), so this example needs an event broker.
```
Comment (nmirasch) [inline serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md]:
```suggestion
A callback is a state that invokes an action and waits for an event (an event that will be eventually fired by the external service notified by the action), so this example needs an event broker.
```

File: serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+This example consist of a callback state that waits for an event arriving on wait channel. Its action publish an event on resume channel. The event published on resume channnel is modified and republished into the wait channel by `PrintService`, which simulates an external service.
```
Comment (nmirasch) [inline serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md]:
```suggestion
This example consists of a callback state that waits for an event arriving on the wait channel. Its action is to publish an event on resume channel. The event published on resume channel is modified and republished into the wait channel by `PrintService`, which simulates an external service. 
```

File: serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+1. Open a Terminal
+2. Go to docker-compose folder
+3. Run ```docker-compose up```
+
+```bash
+cd docker-compose && docker-compose up
+```
+
+Tip: If you get permission denied while creating the postgres container, consider to use SELinux context.
```
Comment (nmirasch) [inline serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md]:
```suggestion
Tip: If you get permission denied while creating the postgres container, consider using SELinux context.
```

**Comment by cristianonicolai** [comment]
@nmirasch fixes applied, thanks

**Comment by cristianonicolai** [comment]
jenkins run quarkus-lts

File: serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
+# Kogito Serverless Workflow - Data Index Example
+
+## Description
+
+This example contains a simple workflow service that demonstrates how to use Data Index Addon as part of the Kogito runtime. 
+A callback is a state that invokes an action and waits for an event (an event that will be eventually fired by the external service notified by the action), so this example needs an event broker.
+This example consists of a callback state that waits for an event arriving on the wait channel. Its action is to publish an event on resume channel. The event published on resume channel is modified and republished into the wait channel by `PrintService`, which simulates an external service. 
+The service is described using JSON format as defined in the 
+[CNCF Serverless Workflow specification](https://github.com/serverlessworkflow/specification).
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md]:
```suggestion
This example contains a simple workflow service that demonstrates how to use Data Index Addon as part of the Kogito runtime. 
A callback is a state that invokes an action and waits for an event (an event that will be eventually fired by the external service notified by the action), so this example needs an event broker.
This example consists of a callback state that waits for an event arriving on the `wait` channel. Its action is to publish an event on the `resume` channel. The event published on the `resume` channel is modified and republished into the `wait` channel by `PrintService`, which simulates an external service. 
The service is described using JSON format as defined in the 
[CNCF Serverless Workflow specification](https://github.com/serverlessworkflow/specification).
```

File: serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+### Kafka
+
+This quickstart requires an Apache Kafka to be available and by default expects it to be on default port and localhost.
+
+* Install and Startup Kafka Server / Zookeeper
+
+https://kafka.apache.org/quickstart
+
+To publish and consume the event, topic "move" is used.
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md]:
```suggestion
To publish and consume the event, the topic `move` is used. 
```

File: serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+docker-compose up
+```
+
+In this way a container for Kafka will be started on port 9092.
+
+### PostgreSQL
+
+This example also requires persistence with a PostgreSQL server.
+
+Configuration for setting up the connection can be found in [applications.properties](src/main/resources/application.properties) file, which
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md]:
```suggestion
The configuration for setting up the connection can be found in [applications.properties](src/main/resources/application.properties) file, which
```

File: serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+### Prerequisites
+ 
+You will need:
+  - Java 11+ installed
+  - Environment variable JAVA_HOME set accordingly
+  - Maven 3.8.1+ installed
+  - Docker and Docker Compose to run the required example infrastructure.
+
+When using native image compilation, you will also need: 
+    - GraalVM 20.3+ installed
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md]:
```suggestion
    - GraalVM 22.2+ installed
```

File: serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+
+### Compile and Run in Local Dev Mode
+
+```sh
+mvn clean package quarkus:dev
+```
+
+### Start infrastructure services
+
+You should start all the services before you execute any of the **Data Index** example, to do that please execute:
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md]:
```suggestion
You should start all the services before you execute any of the **Data Index** example. To do that please execute:
```

File: serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+
+1. Open a Terminal
+2. Go to docker-compose folder
+3. Run ```docker-compose up```
+
+```bash
+cd docker-compose && docker-compose up
+```
+
+Tip: If you get permission denied while creating the postgres container, consider using SELinux context.
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md]:
```suggestion
TIP: If you get a `permission denied` error while creating the postgresql container, consider using SELinux context.
```

File: serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+```yaml
+    - ./sql:/docker-entrypoint-initdb.d:Z
+```
+
+Once all services bootstrap, the following ports will be assigned on your local machine:
+
+- PostgreSQL: 5432
+- Kafka: 9092
+- PgAdmin: 8055
+- serverless-workflow-service :8080
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md]:
```suggestion
- serverless-workflow-service: 8080
```

File: serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+Once started you can simply stop all services by executing the ```docker-compose -f docker-compose.yml stop```.
+
+All created containers can be removed by executing the ```docker-compose -f docker-compose.yml rm```.
+
+
+### Compile and Run using Local Native Image
+Note that this requires GRAALVM_HOME to point to a valid GraalVM installation
+
+```sh
+mvn clean package -Pnative
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md]:
```suggestion
mvn clean package -Dnative
```

**Comment by cristianonicolai** [comment]
@radtriste suggestions applied, thanks

**Review by nmirasch — APPROVED** [review]

**Review by radtriste — APPROVED** [review]

**Pull request merged**